operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,174 @@
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// +kubebuilder:validation:Enum=BatchJob;RayJob;RayCluster;JobSet;MPIJob;PaddleJob;PytorchJob;TFJob;XGBoostJob;AppWrappers;Pod;Deployment;StatefulSet;LeaderWorkerSet
+type KueueIntegrations string
+
+const (
+	BatchJob        KueueIntegrations = "BatchJob"
+	RayJob          KueueIntegrations = "RayJob"
+	RayCluster      KueueIntegrations = "RayCluster"
+	JobSet          KueueIntegrations = "JobSet"
+	MPIJob          KueueIntegrations = "MPIJob"
+	PaddeJob        KueueIntegrations = "PaddeJob"
+	PyTorchJob      KueueIntegrations = "PyTorchJob"
+	TFJob           KueueIntegrations = "TFJob"
+	XGBoostJob      KueueIntegrations = "XGBoostJob"
+	AppWrappers     KueueIntegrations = "AppWrappers"
+	Pod             KueueIntegrations = "Pod"
+	Deployment      KueueIntegrations = "Deployment"
+	StatefulSet     KueueIntegrations = "StatefulSet"
+	LeaderWorkerSet KueueIntegrations = "LeaderWorkerSet"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length. The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// The optional prefix and the name are separate by a forward slash (/).
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Group string `json:"group"`
+	// resource of external framework
+	// must be a valid qualified name consisting of a lower-case alphanumeric string,
+	// and hyphens of at most 63 characters in length. The name must start and end with an alphanumeric character.
+	// The name may be optionally prefixed with a subdomain consisting of lower-case alphanumeric characters,
+	// hyphens and periods, of at most 253 characters in length.
+	// Each period separated segment within the subdomain must start and end with an alphanumeric character.
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.qualifiedName().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
```

Comment:
I'm not certain, but I don't think that resources can be full RFC1123 subdomain names. My understanding is that they would end up being limited to a [DNS label](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names). 

@JoelSpeed would probably know more clearly than I.

Reply:
I think that's correct yes, resource is a DNS label which is a slightly different format validation